broker/nats.go

```diff
@@ -59,7 +60,10 @@ func New(url string) (Nats, error) {
 	}, nil
 }
 
-func (b broker) Publish(_ context.Context, _ string, msg Message) error {
+func (b broker) Publish(_ context.Context, token string, msg Message) error {
+	// Add publisher timestamp
+	msg.Time = time.Now().Unix()
```

Comment:
This should be added at the time of message creation. I agree that it's simpler to create timestamp here rather than in each adapter separately, but the Publisher should not modify the message.

Reply:
I don't like the name also `Time`, if you follow @dusanb94 logic that adapter should add this time on msg creation then `Created` timestamp is more logical to me. Even if you put `Published` timestamp here as @manuio did, for me it is OK, even if I agree with @dusanb94, its a bit dirty if we modify msg here, but not to dirty for me. But `Time` timestamp says nothing to me :)

Reply:
Messages must be timestamped in protocol adapters, not here.

Agreed with @nmarcetic that `time` is a bad choice for a field name. I would expect `timestamp`, as traditionally used `created` in this context is not really adequate.